### ISSUE TITLE
publicOnly ignored when inside public functions that are missing jsdoc

### ISSUE THREAD

**Issue by FTWinston** [gajus/eslint-plugin-jsdoc#812]
When the `publicOnly` option is specified, the rule seems to be triggered for local functions within public members, if the public member itself lacks jsdoc. This is similar to #654, but with an extra caveat.

## Expected behavior
Local (nested) functions should not trigger a linter failure if not documented, regardless of whether the enclosing public function is missing required jsdoc.

## Actual behavior
Local (nested) functions trigger a linter failure if not documented, but only when the enclosing public function is missing required jsdoc.

## ESLint Config
```js
module.exports = {
	plugins: ['jsdoc'],
	rules: {
		'jsdoc/require-jsdoc': [
			'error',
			{
				publicOnly: true,
				require: {
					ClassDeclaration: true,
					ClassExpression: true,
					ArrowFunctionExpression: true,
					FunctionDeclaration: true,
					FunctionExpression: true,
					MethodDefinition: true,
				},
			},
		],
	},
};
```

## ESLint sample

The `inner` function here triggers the rule:

```js
export function outer() {
    function inner() {
        console.log('foo');
    }

    inner();
}
```

But the `inner` function here does not:

```js
/**
 *
 */
export function outer() {
    function inner() {
        console.log('foo');
    }

    inner();
}
```

## Environment

- Node version: 16.13.0
- ESLint version: 7.32.0
- `eslint-plugin-jsdoc` version: 37.2.0

**Comment by FTWinston** [comment]
This is a bit of an edge case, and only really relevant when applying eslint-plugin-jsdoc to code that didn't already use it.

**Comment by brettz9** [comment]
The applying of eslint-plugin-jsdoc to code that didn't already use it is handled within the `jsdoc/require-jsdoc` rule specifically.

Since the other example *should* trigger the rule since the main export is missing jsdoc, it appears you mean that what is happening is that we are misreporting the line number as coming from the inner function when it should be reporting the outer public one. Is that correct?

**Comment by brettz9** [comment]
Ah testing locally, it appears instead that we are reporting both the inner and outer in such a case when we should only be reporting one...

**Comment by FTWinston** [comment]
Yes, that's what I'd meant. Sorry for the confusion!

Outer correctly reports an error, but inner also reports an error, incorrectly.

**Comment by brettz9** [comment]
This is somewhat of a hackish solution, but will hopefully do the trick.

**Comment by FTWinston** [comment]
Wow that was fast, thanks!

That's fixed the problem with nested function declarations, but an equivalent problem still occurs with nested arrow functions.

i.e. with `inner` here:
```js
export const outer = () => {
    const inner = () => {
        console.log('foo');
    };

    inner();
};
```

and here:
```js
export function outer() {
    const inner = () => {
        console.log('foo');
    };

    inner();
};
```

but **not** here:
```js
export const outer = () => {
    function inner() {
        console.log('foo');
    }

    inner();
};
```

My apologies for not mentioning arrow functions previously. Would you like me to raise a separate ticket?

**Comment by FTWinston** [comment]
Superb, thanks!
